enterprise/server/quota/quota_manager_test.go

```diff
@@ -314,6 +317,68 @@ func TestBucketRowFromMap(t *testing.T) {
 	}
 }
 
+func TestConcurrentBucketAccess(t *testing.T) {
+	env := testenv.GetTestEnv(t)
+	testUsers := testauth.TestUsers("US001", "GR001")
+	env.SetAuthenticator(testauth.NewTestAuthenticator(testUsers))
+	ctx := testauth.WithAuthenticatedUserInfo(context.Background(), testUsers["US001"])
+
+	adb, err := authdb.NewAuthDB(env, env.GetDBHandle())
+	require.NoError(t, err)
+	env.SetAuthDB(adb)
+	udb, err := userdb.NewUserDB(env, env.GetDBHandle())
+	require.NoError(t, err)
+	env.SetUserDB(udb)
+
+	flags := map[string]memprovider.InMemoryFlag{
+		bucketQuotaExperimentName: {
+			State:          memprovider.Enabled,
+			DefaultVariant: "custom",
+			Variants: map[string]any{
+				"custom": map[string]any{
+					"rpc:/test.Service/Method": map[string]any{
+						"maxRate": map[string]any{
+							"numRequests": int64(100),
+							"periodUsec":  int64(60000000),
+						},
+						"maxBurst": int64(50),
+					},
+				},
+			},
+		},
+	}
+	provider := memprovider.NewInMemoryProvider(flags)
+	domain := t.Name()
+	require.NoError(t, openfeature.SetNamedProviderAndWait(domain, provider))
+
+	fp, err := experiments.NewFlagProvider(domain)
+	require.NoError(t, err)
+	env.SetExperimentFlagProvider(fp)
+
+	qm, err := newQuotaManager(env, createTestBucket)
+	require.NoError(t, err)
+
+	const numGoroutines = 10
+	const numIterations = 100
+
+	done := make(chan bool)
+	for i := 0; i < numGoroutines; i++ {
+		keyID := i
+		go func() {
+			defer func() { done <- true }()
+			key := fmt.Sprintf("key%d", keyID)
+			for j := 0; j < numIterations; j++ {
+				_ = qm.loadQuotasFromFlagd(ctx, key, "rpc:/test.Service/Method")
```

Comment:
do you need the "_ = " here or can you just call the method directly (same below)